overlay/default.nix

```diff
@@ -12,6 +12,48 @@ in
   # Uses sources.mix-to-nix and sources.gitignore
   nixUnstable = inputs.nix.defaultPackage.${final.system};
 
+  benchwrapper = prev.writeShellScriptBin "benchwrapper" ''
+    set -euo pipefail
+
+    # number of cores to run the wrapped program on
+    test_cpus=2
+
+    sctl=${prev.systemd}/bin/systemctl
+
+    #############
+
+    ncpus=$(( $(${prev.util-linux}/bin/lscpu -e | wc -l) - 1 ))
+
+    fullrange="0-$(( $ncpus - 1 ))"
+    sysrange="0-$(( $ncpus - $test_cpus - 1 ))"
+    shieldrange="$(( $ncpus - $test_cpus ))-$(( $ncpus - 1 ))"
+
+    cleanup() {
+      echo "called cleanup"
+      # restore full core utilization for all services
+      $sctl set-property --runtime system.slice AllowedCPUs="$fullrange"
+      $sctl set-property --runtime user.slice AllowedCPUs="$fullrange"
```

Comment:
I think that `cleanup` should also do `rm /sys/fs/cgroup/shield`

Reply:
systemd-run solves this issue too, there is no shield slice after the invocation